core/iwasm/fast-jit/fe/jit_emit_numberic.c

```diff
@@ -1000,34 +1030,14 @@ compile_int_rotl(JitCompContext *cc, JitReg left, JitReg right, bool is_i32)
 
     left = mov_left_to_reg(cc, is_i32, left);
 
-    if (is_i32) {
-        tmp = jit_cc_new_reg_I32(cc);
-        shl_res = jit_cc_new_reg_I32(cc);
-        shr_res = jit_cc_new_reg_I32(cc);
-        res = jit_cc_new_reg_I32(cc);
-    }
-    else {
-        tmp = jit_cc_new_reg_I64(cc);
-        shl_res = jit_cc_new_reg_I64(cc);
-        shr_res = jit_cc_new_reg_I64(cc);
-        res = jit_cc_new_reg_I64(cc);
-    }
-
-    /* 32/64 - rhs */
-    GEN_INSN(SUB, tmp, is_i32 ? NEW_CONST(I32, 32) : NEW_CONST(I64, 64), right);
+    res = is_i32 ? jit_cc_new_reg_I32(cc) : jit_cc_new_reg_I64(cc);
 #if defined(BUILD_TARGET_X86_64) || defined(BUILD_TARGET_AMD_64)
     GEN_INSN(MOV, is_i32 ? ecx_hreg : rcx_hreg, right);
-    GEN_INSN(SHL, shl_res, left, is_i32 ? ecx_hreg : rcx_hreg);
-
-    GEN_INSN(MOV, is_i32 ? ecx_hreg : rcx_hreg, tmp);
-    GEN_INSN(SHRU, shr_res, left, is_i32 ? ecx_hreg : rcx_hreg);
-
-    GEN_INSN(OR, res, shl_res, shr_res);
+    GEN_INSN(MOV, res, left);
```

Comment:
it should work well, but why `mov` firstly?

Reply:
Yes, there are bugs in codegen shift_r_r_to_r_i32/shift_r_r_to_r_i64, see the modify in codegen, we should not modify the source register (shl reg_no1_src cl will change reg_no1_src). I didn't find it before, so modify here. We can restore it after codegen issue is fixed.

Reply:
Restored, done.